src/sync/impl/sync_client.hpp

```diff
@@ -42,6 +46,24 @@ struct SyncClient {
         client.run();
     }) // Throws
     {
+#if REALM_PLATFORM_APPLE
```

Comment:
It looks like that reachability observing on ObjectStore level is only possible on Apple platforms, so there wont be any other implementations. But if you think it's reasonable I'm happy to do that.